onnxruntime/test/perftest/command_args_parser.cc

```diff
@@ -41,8 +41,8 @@ namespace perftest {
       "\t-p [profile_file]: Specifies the profile name to enable profiling and dump the profile data to the file.\n"
       "\t-s: Show statistics result, like P75, P90.\n"
       "\t-v: Show verbose information.\n"
-      "\t-x [intra_op_num_threads]: Sets the number of threads used to parallelize the execution within nodes, A value of 0 means ORT will pick a default. Must >=0.\n"
-      "\t-y [inter_op_num_threads]: Sets the number of threads used to parallelize the execution of the graph (across nodes), A value of 0 means ORT will pick a default. Must >=0.\n"
+      "\t-x [intra_op_num_threads]: Sets the number of threads used to parallelize the execution within nodes, A value of 0 means ORT will pick a default. Must >=0. If OpenMP is enabled, this configuration will be ignored.\n"
```

Comment:
>-x [intra_op_num_threads] [](start = 9, length = 25)

Can we change -x to call omp_set_num_threads() when building with OpenMP?

Reply:
currently this is controlled by env var `OMP_NUM_THREADS`. if -x has different value to `OMP_NUM_THREADS`, what is the expected behavior? I think this may cause confusing

Reply:
I believe set_omp_num_threads would override the env then, which is what user would expect if OpenMP is mainly used for intra node or data parallelism. Otherwise, we need to ask user to use different ways to specify data parallelism #threads.

---
In reply to: [350977758](https://github.com/microsoft/onnxruntime/pull/2485#discussion_r350977758) [](ancestors = 350977758)

Reply:
There is one tricky part of the current behavior:

if OpenMP is not enabled, if you set `-x 8`, it means you will have 9 threads (main + 8 workers)
if OpenMP is enabled, if you set `OMP_NUM_THREADS=8`, it means you will have 8 threads in total (ie. main + 7 workers)

this cause the problem: if we want to extend `-x`, the behavior will be inconsistent.

Reply:
Unless we unify the definition of "numThreads" ( number of worker threads? or number of threads in total? ), I would prefer to keep the current status

Reply:
Even if -x means different things when using OpenMP vs. not using OpenMP, I'd prefer to have -x to set OMP thread count, because it's easier to control than setting env and prevents accidentally picking up previously set env, especially in Windows.

---
In reply to: [351000042](https://github.com/microsoft/onnxruntime/pull/2485#discussion_r351000042) [](ancestors = 351000042)

Reply:
You cannot expect that everyone fully understand this tricky behavior, and it's not necessarily have to understand. It can't be easier to make mistakes -- create threadpool with the wrong number of threads without being aware of it.

I would even prefer to add a new flag rather than `-x`, for example, `--omp-num-threads` or anything you name it, but not combining it with `-x` to make it multi-behavior.

Reply:
Sounds good to me, so we just disable -x in OpenMP build.

---
In reply to: [351019314](https://github.com/microsoft/onnxruntime/pull/2485#discussion_r351019314) [](ancestors = 351019314)